trite-osprey/src/List.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import Cart from './Cart';
+
+class List extends React.Component{
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      beers : [],
+      currentBeer: null,
+      searchString: '',
+      visible: false,
+      active: this.props.active || 0,
+      beerList: [],
+      totalPrice: 0,
+      currentBeerObjects: []
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  componentWillMount() {
+    this.getAllBeers()
+  }
+
+  getAllBeers() {
+    fetch('http://localhost:5000/beers', {
+      method: 'get',
+    })
+    .then(response => response.json())
+    .then(results => {
+      this.setState({
+        beers: results
+      })
+    })
+  }
+
+  handleChange(event) {
+    this.setState( { searchString: event.target.value} )
+  }
+
+  hide() {
+    this.setState({visible: false})
+  }
+
+  handleClick(event, index) {
+    this.setState({visible: true, active: index, currentBeer: this.state.beers[index] })
+  }
+
+  addToCart() {
+    let { currentBeer, beerList, currentBeerObjects } = this.state
+    currentBeerObjects.push(currentBeer)
+    beerList.push(currentBeer.beername)
+
+    let priceArray = [];
+    Object.keys(currentBeerObjects).forEach(function(key) {
+      let val = currentBeerObjects[key]["price"];
+      priceArray.push(val);
+    });
+
+    let cartPrice = priceArray.reduce((a,b) => a + b)
+
+    this.setState({
+      totalPrice: cartPrice,
+      beerList,
+      currentBeerObjects
+    })
+
+  }
+
+  removeFromCart() {
+    let { currentBeer, beerList, currentBeerObjects } = this.state
+
+    let id = currentBeer.id;
+    for (var i = 0; i < currentBeerObjects.length; i++) {
+        if (currentBeerObjects[i].id === id) {
+            currentBeerObjects.splice(i, 1);
+            break;
+        }
+    }
+
+    let priceArray = [0];
+    Object.keys(currentBeerObjects).forEach(function(key) {
+      let val = currentBeerObjects[key]["price"];
+      priceArray.push(val);
+    });
+
+    let beerName = currentBeer.beername;
```

Comment:
Dont modify the object you have in state. Make new objects and set them as the new state value. 

`beerList = beerList.filter(beerName => beerName !== currentBeer.beername)`

but the fact that you have `currentBeerObjects` and `beerList` suggests you have duplicate data in your state. Try to reduce your state as much as possible and derive what you need within your render function.